test/unit/adapters/http.js

```diff
@@ -260,6 +260,41 @@ describe('supports http with nodejs', function () {
     });
   });
 
+  it('should support max content length for redirected', function (done) {
+    var str = Array(100000).join('ж');
+
+    server = http.createServer(function (req, res) {
+      var parsed = url.parse(req.url);
+
+      if (parsed.pathname === '/two') {
+        res.setHeader('Content-Type', 'text/html; charset=UTF-8');
+        res.end(str);
+      } else {
+        res.setHeader('Location', '/two');
+        res.statusCode = 302;
+        res.end();
+      }
+    }).listen(4444, function () {
+      var success = false, failure = false, error;
+
+      axios.get('http://localhost:4444/one', {
+        maxContentLength: 2000
+      }).then(function (res) {
+        success = true;
+      }).catch(function (err) {
+        error = err;
+        failure = true;
+      });
+
+      setTimeout(function () {
+        assert.equal(success, false, 'request should not succeed');
+        assert.equal(failure, true, 'request should fail');
+        assert.equal(error.message, 'maxContentLength size of 2000 exceeded');
+        done();
+      }, 100);
```

Comment:
we can just move the assertions to the catch and exercise the same behavior? to make the specs run faster since people will have to wait the extra 100ms each time to run the test suite

Reply:
@yasuf Good suggestion. I just copied a similar case and changed a little for redirection. Opened #2699 to do that separately.

Reply:
@yasuf Can we merge it now?